### DISCUSSION TITLE
Error rendering Quarto website from Dropbox

### DISCUSSION THREAD

**Discussion by eteitelbaum** [quarto-dev/quarto-web#295] [Q&A]
When I try to render a Quarto website saved in Dropbox I often get the following message: 

```
ERROR: The process cannot access the file because it is being used by another process. (os error 32), remove 'C:\Users\emman\Dropbox\Website\quarto-test2\index_files'
```

If I delete the `index_files` folder sometimes it works but sometimes I get the message again. 

I do not get this message if I store the website in my `Documents` folder outside of Dropbox. 

Am I right in thinking that this error is related to Dropbox?

**Comment by dragonstyle** [comment]
A guess is that the files are being synchronized by Dropbox as they are written by Quarto and perhaps that prevents Quarto from moving or deleting files. I'm not 100% sure of this though we don't have a general issue that we're aware of with 'file in use' type errors. You could try pausing Dropbox sync and see whether that resolves the issue (that would at least confirm it is sync related).

**Reply by eteitelbaum** [reply]
Yes, that was the source of the problem. When I pause Dropbox syncing it works. Should I post it as an issue?

**Reply by rgaiacs** [reply]
@eteitelbaum Thanks for the question. This is not an issue in Quarto but a limitation of synchronising files across machines.

**Reply by eteitelbaum** [reply]
@rgaiacs Thanks. Are you saying this is an issue with Dropbox? I only experience it when using Quarto.

**Reply by rgaiacs** [reply]
@eteitelbaum Issues are different from limitations. Limitations are by design, for example humans can't fly because they don't have wings. Issues are different behaviour against the design, for example bird can't fly because it's wing is broken.

Have two files accessing the same file at the same time can be a little problematic and is best to avoid it. My understanding from your description is that after you render your project with Quarto, Dropbox will see the changes in the files inside `index_files` and create a temporary lock to the files inside `index_files` until Dropbox backup the files. Before Dropbox is able to backup the files and unlock them, you try to render your project again. This time, Quarto try to write the new files but it fails because the files are locked by Dropbox.

Dropbox design is to lock the files. The majority of the time you will not notice it. Unfortunately, sometimes you will get the error. Quarto design is to respect the file lock.

**Reply by eteitelbaum** [reply]
@rgaiacs Thanks for your response. 

Actually, I can't render the files at all when Dropbox is on. So it is not a case of rendering sometimes and not others. 

To take your analogy one step further, humans can't fly so they built airplanes. Similarly, maybe Quarto can be engineered to play nice with Dropbox. It seems like it should be possible given that I don't have this problem when I render with R Markdown, but I admit that I am not a software engineer. 

If you tell me that it is impossible to render a Quarto document to a Dropbox folder, then I would say that this should be noted in the documentation. Lots of people use Dropbox. Personally, I use it to back up all of my files. So I will have to change my workflow to either turn off Dropbox or save my Quarto docs outside of Dropbox and then find some other way to back them up.

**Reply by rgaiacs** [reply]
> Actually, I can't render the files at all when Dropbox is on. So it is not a case of rendering sometimes and not others.

This sounds strange to me. I expect that you would be able to at least render once. For clarification, we have

```
$ quarto help render

  Usage:   quarto render [input] [args...]
  Version: 0.9.649                        
                                          

  Description:

    Render input file(s) to various document types.                                      

  Commands:

    help  [command]  - Show this help or the help of a sub-command.

  Examples:

    Render Markdown:    quarto render document.qmd
```

and

```
$ quarto help preview

  Usage:   quarto preview [file] [args...]
  Version: 0.9.649                        
                                          

  Description:

    Render and preview a document or website project.                                                   
                                                                                                        
    Automatically reloads the browser wheninput files or document resources (e.g. CSS) change.          
                                                                                                        
    For website preview, the most recent execution results of computational documents are used to render
    the site (this is to optimize startup time). If you want to perform a full render prior to          
    previewing pass the --render option with "all" or a comma-separated list of formats to render.      
                                                                                                        
    For document preview, input file changes will result in a re-render (pass --no-watch to prevent).   
                                                                                                        
    You can also include arbitrary command line arguments to be forwarded to quarto render.             

  Examples:

    Preview document:                                   quarto preview doc.qmd
```

When you click the `Render` button in the file editor panel, RStudio will use `quarto preview`. I don't know if using `quarto render` will have any difference.

> Similarly, maybe Quarto can be engineered to play nice with Dropbox.

You are not the first one to have a similar problem. See https://community.rstudio.com/t/mostly-resolved-dropbox-box-sync-google-drive-etc/6098.

> Similarly, maybe Quarto can be engineered to play nice with Dropbox.

Or maybe Dropbox could be engineered to not lock the files. ;-)

> It seems like it should be possible given that I don't have this problem when I render with R Markdown

It might be related with the `quarto preview` vs `quarto render` that I mentioned earlier. My experience is that in RStudio, when I'm working in a `.Rmd` file, the `knit` button in will convert the file but not start any web server.

> If you tell me that it is impossible to render a Quarto document to a Dropbox folder, then I would say that this should be noted in the documentation.

Quarto is a very new tool and we are still discovering it's limitations. Thanks to share with us the current limitation with Dropbox.

**Reply by cdriveraus** [reply]
I have the same issue, it's a bit frustrating. Rendering via Rstudio, freshly opened, no changes made recently. I understand it's not explicitly a quarto 'problem', but quarto seems a little unique in this limitation - I guess it's somehow changing a file, dropbox is seeing this and starts to sync, then the quarto render process wants to access the file again very quickly? Maybe the quarto process needs to put a lock on the important files until it is finished, somehow?
Others with the same issue: https://stackoverflow.com/questions/73324250/quarto-persistent-problems-trying-to-render-preview
